request-manager/request/resumer.go

```diff
@@ -0,0 +1,606 @@
+// Copyright 2018, Square, Inc.
+
+package request
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	myconn "github.com/go-mysql/conn"
+
+	jr "github.com/square/spincycle/job-runner"
+	"github.com/square/spincycle/proto"
+	"github.com/square/spincycle/request-manager/db"
+)
+
+type Resumer interface {
+	// Run periodically attempts to resume any Suspended Job Chains stored in
+	// the database, and also cleans up any SJCs that get into a bad state. Run
+	// blocks until the Resumer is shut down by closing shutdownChan.
```

Comment:
`shutdownChan` is an implementation detail. Better to document it at the object that implements this interface.